### ISSUE TITLE
Flatten waves without effecting physics.

### ISSUE THREAD

**Issue by holdingjason** [huwb/crest-oceanrender#190]
Ok not sure what the best way to do this is but here is the issue.

So similar to masking out a wave ie so that we do not see water etc inside of the row boat we will probably also want to flatten waves around that same area.  This would be so that big waves do not go over the top of the mesh/boat.  We would not however want this to effect in anyway the actual wave heights/water height used in calculating physics/boat probes etc.

Right now we have the wake distortion guy that does cause the waves etc to squish down etc that is used to create wave wakes for the boat examples however this is a physical distortion which does effect how the boat reacts.  So say you put it at the front in order to do something like I was talking about earlier it would cause the boat to react a bounce for instance which is not what I would want.

Basically the masking only goes so far.  Works great on mostly flat ocean and in a confined area that you can catch angles around like the row boat, does not work on larger waves that would go around or over the mask or if you have a flat deck etc.  So would seem we need someway to basically say that the area this mask covers could also flatten the waves but only at that spot. 

I think this was discussed somewhat with the submarine issue but need to dig that up.

**Comment by huwb** [comment]
You can render whatever you want into the wave data, so you could locally flatten out the waves or do something there.

I'm not sure about doing this without affecting physics. The two are kinda connected together. I suppose if you set the collision source (in the Animated Waves Settings) to be CPU, the cpu water height queries would not take into account any custom stuff you've rendered.

I'm not sure if these really help, ill think about if i have other ideas.

**Comment by holdingjason** [comment]
Thanks.  Yeah same.  Really trying to understand what it is I want to accomplish here so maybe a better way to do it or perhaps should just avoid these kind of waves all together (simple solution).   Perhaps it effecting physics is also ok since would only do it if the waves were over the height of the bow or something.

**Comment by huwb** [comment]
I reckon you could draw a black blob into the smaller lods which have the choppier waves, to flatten them out. This would only affect the smaller wavelengths. This may still affect the physics of your using displacement texture read backs, but you are probably not using the small wavelengths for physics.

There should be an option on the register anim waves input script to select particular wavelengths to render into. If you specify a preference here it will render before the combine pass, so you can really manipulate particular wave octaves while leaving others untouched.

**Comment by holdingjason** [comment]
Cool.  Ok will play with that and see how it goes.

**Comment by huwb** [comment]
Hi there, did you have any luck with this, out of curiosity? I'll close as its been inactive for > 4 weeks. We can reopen if needed.

**Comment by holdingjason** [comment]
Sorry got pulled into other iteration issues so need to try and get some time to get back to this.  Will still check it out and let you know.
